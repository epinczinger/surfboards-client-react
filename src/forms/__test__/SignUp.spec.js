import React from "react";
import renderer from "react-test-renderer";
import SignUp from "../../forms/SignUp";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../reducers";

describe("The SignUp component", () => {
  it("renders as expected matching snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <SignUp />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
