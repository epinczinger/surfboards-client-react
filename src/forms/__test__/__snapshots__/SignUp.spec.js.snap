// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SignUp component renders as expected matching snapshot 1`] = `
<div
  className="flex flex-row items-center mx-auto pt-4"
>
  <div
    className="w-11/12 w-full max-w-xs mx-auto"
  >
    <h2
      className="sign-title text-3xl text-gray-700 font-medium sm:text-5xl text-center mb-4"
    >
      Sign Up!
    </h2>
    <form
      className="bg-white shadow-md text-xl rounded px-8 pt-6 pb-8 mb-4"
    >
      <label
        className="pb-2 block text-gray-700 font-bold mb-2"
        htmlFor="email"
      >
        Email
        <input
          className="mt-2 text-base sm:text-lg h-11 shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          name="email"
          onChange={[Function]}
          placeholder="example@mail.com"
          type="email"
          value=""
        />
      </label>
      <label
        className="pb-2 block text-gray-700 font-bold mb-2"
        htmlFor="password"
      >
        Password
        <input
          className="mt-2 h-11 text-base sm:text-lg shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          name="password"
          onChange={[Function]}
          onKeyUp={[Function]}
          type="password"
          value=""
        />
      </label>
      <label
        className="pb-2 block text-gray-700 font-bold mb-2"
        htmlFor="passwordConfirmation"
      >
        Password Confirmation
        <input
          className="mt-2 h-11 text-base sm:text-lg shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          name="passwordConfirmation"
          onChange={[Function]}
          onKeyUp={[Function]}
          type="password"
          value=""
        />
      </label>
      <a
        href="/sign_in"
        onClick={[Function]}
      >
        <p
          className="text-blue-400 hover:text-blue-700"
        >
          Already have an account?
        </p>
      </a>
      <button
        className="w-full mt-5 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
        onClick={[Function]}
        type="button"
      >
        SignUp
      </button>
    </form>
  </div>
</div>
`;
